tests/bindings/python/test_reset.py

```diff
@@ -29,115 +29,73 @@ def test_set_get_variable(self):
         from libcellml import Reset
-        from libcellml import When
-
-        w1 = When()
-        w2 = When()
-        w3 = When()
-        r = Reset()
-
-        r.addWhen(w1)
-        r.addWhen(w2)
-        r.addWhen(w3)
-        self.assertEqual(3, r.whenCount())
 
-        r.removeAllWhens()
-        self.assertEqual(0, r.whenCount())
+        x = Reset()
+        y = Reset()
+        z = Reset(y)
+        del(x, y, z)
 
-    def test_contains_when(self):
+    def test_set_order(self):
         from libcellml import Reset
-        from libcellml import When
 
-        w1 = When()
-        w2 = When()
-        r = Reset()
-
-        r.addWhen(w1)
-
-        self.assertTrue(r.containsWhen(w1))
-        self.assertFalse(r.containsWhen(w2))
+        x = Reset()
+        x.setOrder(3)
+        self.assertEqual(3, x.order())
 
-    def test_when(self):
+    def test_is_order_set(self):
         from libcellml import Reset
-        from libcellml import When
 
-        w = When()
-        r = Reset()
-
-        r.addWhen(w)
-
-        self.assertEqual(None, r.when(2))
-        self.assertEqual(None, r.when(-4))
-        self.assertNotEqual(None, r.when(0))
+        # bool isOrderSet()
+        x = Reset()
+        self.assertFalse(x.isOrderSet())
+        x.setOrder(2)
+        self.assertTrue(x.isOrderSet())
 
-    def test_take_when(self):
+    def test_unset_order(self):
         from libcellml import Reset
-        from libcellml import When
 
-        w = When()
-        r = Reset()
+        # unsetOrder()
+        x = Reset()
+        x.setOrder(-4)
+        self.assertTrue(x.isOrderSet())
+        x.unsetOrder()
+        self.assertFalse(x.isOrderSet())
 
-        r.addWhen(w)
+    def test_set_get_test_value(self):
+        from libcellml import Reset
 
-        self.assertEqual(1, r.whenCount())
-        self.assertEqual(None, r.takeWhen(2))
+        x = Reset()
+        self.assertEqual("", x.testValue())
 
-        w_returned = r.takeWhen(0)
+        x.setTestValue("A MathML string.")
 
-        self.assertNotEqual(None, w_returned)
-        self.assertEqual(0, r.whenCount())
+        self.assertEqual("A MathML string.", x.testValue())
 
-    def test_replace_when(self):
+    def test_set_get_reset_value(self):
         from libcellml import Reset
-        from libcellml import When
 
-        w1 = When()
-        w1.setCondition("x=exp(y)")
-        w2 = When()
-        w2.setValue("a=2")
-
-        r = Reset()
-
-        r.addWhen(w1)
+        x = Reset()
+        self.assertEqual("", x.resetValue())
 
-        self.assertFalse(r.replaceWhen(3, w2))
-        self.assertEqual("x=exp(y)", r.when(0).condition())
-        self.assertEqual("", r.when(0).value())
+        x.setResetValue("A value MathML string.")
 
-        self.assertTrue(r.replaceWhen(0, w2))
-        self.assertEqual("", r.when(0).condition())
-        self.assertEqual("a=2", r.when(0).value())
+        self.assertEqual("A value MathML string.", x.resetValue())
```

Comment:
Shouldn't we do as we are doing in our C++ tests, i.e. make the set/tested value a constant?

Reply:
Actually, I can see that this is not done in the C++ tests below either... Oh well...

Reply:
I'm pretty sure that the compiler will turn this into a constant array.  I don't think that this is a problem that needs addressing.

Reply:
Agreed, but not my point. I am after consistency in our code.